### DISCUSSION TITLE
Tag Detection & Target Hitting

### DISCUSSION THREAD

**Discussion by KierenSama2** [larics/icuas22_competition#49] [Q&A]
Dear Organizers,

For task 2, I am are aware that an annotated image of the tag must be published to a ROS topic. However I am unsure what the annotation should include... do we just include the x, y, z position of the target in the global frame?

![Screenshot from 2022-04-13 14-51-30](https://user-images.githubusercontent.com/65677723/163438413-f85397cc-2c50-4996-8c2e-9eb6a2f58aaa.png)

Also, when throwing the ball to the target, is there a way to determine whether the target was hit?

Thank you

**Comment by fpetric** [comment]
Hi @KierenSama2, 

annotation can be anything that we can use as a proof that you actually detected the tag (we check all successful runs manually, just to make sure teams did not launch ball randomly). Bounding box would be best, but even the image that you provided as an example is ok. Just make sure to publish it to correct topic, see instructions [here ](https://github.com/larics/icuas22_competition/discussions/43). Also make sure to publish the reconstructed position of the tag. 

With respect to tag detection, make sure to pull new changes from the main repo and from larics_gazebo_worlds, we have introduced new arenas for you to try your solution and also added textures to walls and obstacles that make the detection a bit more difficult. 

When throwing the ball, after you release the ball by publishing to magnet gain topic, you can track the position of the ball (it is published on topic `/red/ball/pose`) and see how close to the position of the tag it comes. We do something similar to measure the precision of the launch when evaluating your solutions.

**Reply by Maik93** [reply]
> With respect to tag detection, make sure to pull new changes from the main repo and from larics_gazebo_worlds, we have introduced new arenas for you to try your solution and also added textures to walls and obstacles that make the detection a bit more difficult.

Are you talking about [those worlds](https://github.com/larics/larics_gazebo_worlds/tree/master/worlds)? How should I change the [session file](https://github.com/larics/icuas22_competition/blob/main/startup/challenge/session.yml) in order to use one of them?

**Reply by fpetric** [reply]
No, the worlds we added are [here](https://github.com/larics/larics_gazebo_worlds/tree/master/models/competition). If you update both larics_gazebo_worlds and this repo (the main competition repo), you will get updated session file that can parse the number of the arena as an argument (see updated README for more details).

**Reply by Maik93** [reply]
cool, thank you!
